pkg/kube/controllers/extendedjob/errand_reconciler.go

```diff
@@ -107,6 +107,11 @@ func (r *ErrandReconciler) Reconcile(request reconcile.Request) (result reconcil
 }
 
 func (r *ErrandReconciler) createJob(ctx context.Context, extJob ejv1.ExtendedJob) error {
+	if extJob.Spec.Template.Labels == nil {
+		extJob.Spec.Template.Labels = map[string]string{}
+	}
+	extJob.Spec.Template.Labels["ejob-name"] = extJob.Name
```

Comment:
I don't understand this code. It modifies the extended job and relies on later code to store this?
Are labels transferred from the extended job to the pod somehow, or how does clean up code find pods with that label?

Could this be moved to be near the actual `Update(extjob)` statement, it's not  obvious from naming and function signature, that the passed in `extJob` is mutated here.

Reply:
About moving - I agree about not wanting to mutate it here.
About the other comments, I don't know what you mean. The label will exist on the pod. That's it - it's required to link it back to the ejob. `job-name` is insufficient.

Reply:
Yes I missed that the label is set on `Template` which is the actual `PodSpec` being for the pod.

Reply:
👍

Reply:
Ah, I see now, we don't actually need to mutate extendjob (not even sure if it does). We just need to pass this modified PodTemplateSpec to Job a few lines below